### ISSUE TITLE
Brickadia Raytracing broken; crashes in identical manner to CP77

### ISSUE THREAD

**Issue by Continous** [HansKristian-Work/vkd3d-proton#891]
[Brickadia](https://brickadia.com/) is a block-building game a la Roblox, Blockland, or Lego. The game is in development, and currently not released.

However, it recently made a duo of releases that implemented DXR support through Unreal Engine. The initial (unstable) build worked under VKD3D 2.4 with all features aside from shadows, however exhibited minor artifacts with reflections. These artifacts may have been an issue with the game or VKD3D, and since the build was not a major release, I won't dwell too much on it.

Currently however, the most up to date Alpha build, which is a release build, simply fails to work at all with DXR. What's notable is that Brickadia exposes all ray tracing settings through console commands **_AND_** certain settings work on the "plate" environment so long as nothing is placed or added.

Specifically the following features are guaranteed to crash;

Shadows
Reflections
Global Illumination
Ambient Occlusion
Translucency

However, ray traced Skylight seems to work initially. If you enable it, it will initiate as-normal and run fine, until any portion of the environment is changed or updated. I believe this is a clear sign of an issue with regards to BVH updates.

I believe Brickadia would be a good candidate for testing VKD3D's compatibility on UE4, as it's a real-world project, the development team is transparent and open, and it provides a fairly controllable and flexible environment.

**Comment by doitsujin** [comment]
Which hardware?

D3D12 is known to be problematic on Nvidia since we cannot emulate certain application bugs correctly without the `VK_VALVE_mutable_descriptor_type` extension, which is why Cyberpunk and probably some other games are unstable.

**Comment by Continous** [comment]
Sorry to never get around to actually posting any logs, but it does now work under the subobject branch.

**Comment by Blisto91** [comment]
@Continous Hi there.
Am i correct in understanding that this issue was resolved?

**Comment by Blisto91** [comment]
@Continous Friendly ping

**Comment by Continous** [comment]
> @Continous Friendly ping

Sorry, I must have forgotten to respond in the mess of moving to a new house!

At the moment, it won't even launch with DX12 so...

**Comment by Blisto91** [comment]
I assume it isn't publicly available until early access release?

**Comment by Continous** [comment]
No, however when I had made this report they did have a semi-open alpha going. I think we can kind of, shelf this, until they do have it in Early Access. I just found it unique and helpful due to its dev team being more available, Linux friendly, and all that while exhibiting similar issues to CP2077.

**Comment by Blisto91** [comment]
@Continous Is this still fine to this day as your last comment suggests?

**Comment by Continous** [comment]
Sorry for the delay again, yes it seems fine.
